2023-09-04 17:44:45,923 - worker_process - INFO - Worker_process started.
2023-09-04 17:44:46,196 - Cell - INFO - site-1.435da4fc-b7f7-4cf9-84db-418362299cf9: created backbone internal connector to tcp://localhost:18714 on parent
2023-09-04 17:44:46,246 - Cell - INFO - site-1.435da4fc-b7f7-4cf9-84db-418362299cf9: created backbone external connector to grpc://192.168.100.3:8102
2023-09-04 17:44:46,247 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00001 ACTIVE tcp://localhost:18714] is starting
2023-09-04 17:44:46,247 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00002 ACTIVE grpc://192.168.100.3:8102] is starting
2023-09-04 17:44:46,247 - FederatedClient - INFO - Wait for client_runner to be created.
2023-09-04 17:44:46,248 - FederatedClient - INFO - Got client_runner after 0.0002224445343017578 seconds
2023-09-04 17:44:46,248 - FederatedClient - INFO - Got the new primary SP: grpc://192.168.100.3:8102
2023-09-04 17:44:46,256 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: client runner started
2023-09-04 17:44:48,863 - Communicator - INFO - Received from secure_project server  (1114754 Bytes). getTask: train time: 0.013583183288574219 seconds
2023-09-04 17:44:48,864 - FederatedClient - INFO - pull_task completed. Task name:train Status:True 
2023-09-04 17:44:48,864 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9]: got task assignment: name=train, id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d
2023-09-04 17:44:48,865 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: invoking task executor LearnerExecutor
2023-09-04 17:44:48,865 - LearnerExecutor - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: Client trainer got task: train
2023-09-04 17:44:48,865 - MNISTLearner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: Client site-1 initialized at 
 /home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../435da4fc-b7f7-4cf9-84db-418362299cf9/app_site-1 
 with args: Namespace(workspace='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/..', startup='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../startup', token='19c5479c-bc3e-43f9-9bcd-5e105ccdd23d', ssid='ebc6125d-0a56-4688-9b08-355fe9e4d61a', job_id='435da4fc-b7f7-4cf9-84db-418362299cf9', client_name='site-1', sp_target='192.168.100.3:8102', parent_url='tcp://localhost:18714', fed_client='fed_client.json', set=['secure_train=true', 'uid=site-1', 'org=nvidia', 'config_folder=config', 'print_conf=True'], local_rank=0, train_config='config/config_train.json', client_config='config/config_fed_client.json', config_folder='config', env='config/environment.json')
2023-09-04 17:44:50,123 - LearnerExecutor - ERROR - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: learner initialize exception: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-09-04 17:44:50,123 - LearnerExecutor - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 74, in initialize
    self.learner.initialize(engine.get_all_components(), fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 155, in initialize
    self._create_datasets()
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 164, in _create_datasets
    self.train_loader = self.train_dataset.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 36, in get_loader
    loader = data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-09-04 17:44:50,123 - ClientRunner - ERROR - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: Processing error from executor LearnerExecutor: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-09-04 17:44:50,124 - ClientRunner - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/private/fed/client/client_runner.py", line 254, in _process_task
    reply = executor.execute(task.name, task.data, fl_ctx, self.task_abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 83, in execute
    self.initialize(fl_ctx)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 77, in initialize
    raise e
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 74, in initialize
    self.learner.initialize(engine.get_all_components(), fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 155, in initialize
    self._create_datasets()
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 164, in _create_datasets
    self.train_loader = self.train_dataset.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 36, in get_loader
    loader = data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-09-04 17:44:50,125 - FederatedClient - INFO - Starting to push execute result.
2023-09-04 17:44:50,128 - Communicator - INFO -  SubmitUpdate size: 575 Bytes. time: 0.003359079360961914 seconds
2023-09-04 17:44:50,129 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9, task_name=train, task_id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d]: result sent to server for task: name=train, id=247ba4ee-91fe-4c2c-bb65-4c126dc32c3d
2023-09-04 17:44:50,813 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9, peer=secure_project, peer_run=435da4fc-b7f7-4cf9-84db-418362299cf9]: received aux request from Server to end current RUN
2023-09-04 17:44:50,813 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: ABORT (RUN) command received
2023-09-04 17:44:50,814 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: ABORT (RUN) requests end run events sequence
2023-09-04 17:44:50,815 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: ABOUT_TO_END_RUN fired
2023-09-04 17:44:50,815 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: END_RUN fired
2023-09-04 17:44:52,131 - ClientRunner - INFO - [identity=site-1, run=435da4fc-b7f7-4cf9-84db-418362299cf9]: run method requests end run events sequence
2023-09-04 17:44:52,146 - FederatedClient - INFO - Shutting down client run: site-1
2023-09-04 17:44:54,534 - MPM - INFO - MPM: Good Bye!
